examples/error-handling/src/with_rejection.rs

```diff
@@ -0,0 +1,36 @@
+use axum::{extract::rejection::JsonRejection, http::StatusCode, response::IntoResponse, Json};
+use axum_extra::extract::WithRejection;
+use chrono::Utc;
+use serde_json::{json, Value};
+use thiserror::Error;
+
+pub async fn handler(
+    WithRejection(Json(value), _): WithRejection<Json<Value>, ApiError>,
+) -> impl IntoResponse {
+    Json(dbg!(value))
+}
+
+#[derive(Debug, Error)]
+pub enum ApiError {
+    #[error(transparent)]
+    JsonExtractorRejection(#[from] JsonRejection),
+}
+
+impl IntoResponse for ApiError {
+    fn into_response(self) -> axum::response::Response {
+        let payload = json!({
+            "message": self.to_string(),
+            "timestamp": Utc::now(),
```

Comment:
I don't think depending on chrono just to add a timestamp is worth it. Hyper already adds a `Date` header anyway.